interfaces/default/js/lazylibrarian.js

```diff
@@ -0,0 +1,466 @@
+        }
+    };
+
+    $('#add_author_name').keyup(function(event){
+        if(event.keyCode == 13){
+            searchAction();
+        }
+    });
+
+    $('#add_author_button').click(function () {
+        searchAction();
+        $(this).attr('disabled', true);
+    });
+
+    $('#add_authorid_button').click(function () {
+        var authorname = $('#add_author_select').find(":selected").attr("author");
+        var value      = $('#add_author_select').val();
+        var valuetext  = $('#add_author_select').find('option:selected').text();
+        $(this).attr('disabled', true);
+        $(this).html('<i class="fa fa-spinner fa-spin fa-inverse fa-fw"></i> Please wait...');
+
+        if($('#add_author_book').find('option:selected').val() === 'authorId') {
+            // Add author only
+            addAuthor(authorname, 'authorId', valuetext, function(){
+                $('#add_author_name').val('');
+                cancelAddAuthor();
+            });
+        }else{
+            // Add author then book
+            addAuthor(authorname, 'authorId', valuetext, function(){
+                addAuthor(value, 'bookId', valuetext, function(){
+                    $('#add_author_name').val('');
+                    cancelAddAuthor();
+                });
+            });
+        }
+    });
+
+    $('#cancel_author_button').click(function () {
+        cancelAddAuthor();
+    });
+
+    $('.lazylibrarian_forceprocess').click(function(e) {
+        e.preventDefault();
+        Postprocess();
+    })
+    
+});
+
+function beginRefreshAuthor(authorId) {
+    var $div = $('div').html('Refreshing author');
+    var $buttons = {
+        'Refresh': function () {
+            beginRefreshAuthor(authorId);
+        }
+    }
+
+    showModal('Refresh author?', $div, $buttons);
+}
+
+function refreshAuthor(authorId) {
+
+    $.ajax({
+        url: WEBDIR + 'lazylibrarian/RefreshAuthor',
+        type: 'post',
+        data: {'authorId': authorId},
+        dataType: 'json',
+        success: function (result) {
+
+        },
+        error: function (req) {
+            console.log('error refreshing author');
+        }
+    })
+}
+
+function searchForAuthor(name, type) {
+    $.ajax({
+        url: WEBDIR + 'lazylibrarian/SearchForAuthor',
+        type: 'get',
+        data: {'name': name,
+                'searchtype': type},
+        dataType: 'json',
+        timeout: 40000,
+        success: function (result) {
+            if (!result || result.length === 0) {
+                $('#add_author_button').attr('disabled', false);
+                return;
+            }
+            // remove any old search
+            $('#add_author_select').html('');
+
+            if (type == 'authorId') {
+                $.each(result, function (index, item) {
+                    var option = $('<option>')
+                    .attr('value', item.authorid)
+                    .attr('author', item.authorname)
+                    .html(item.authorname);
```

Comment:
Consider adding author date or something so its easier to pick the correct author. I searched for dan brown while testing and i got alot results.